File: archiso/airootfs/usr/share/churros/preferences/widgets/sidebar_item.py
import os
import gi

gi.require_version("Gtk", "4.0")

from gi.repository import Gtk


ICON_DIR = os.path.join(
    os.path.dirname(__file__),
    "..",
    "assets",
    "icons"
)


class SidebarItem(Gtk.Button):

    def __init__(
        self,
        page,
        icon,
        title
    ):

        super().__init__()

        self.page = page

        self.add_css_class(
            "sidebar-item"
        )

        content = Gtk.Box(
            orientation=Gtk.Orientation.HORIZONTAL,
            spacing=12
        )

        content.set_margin_top(10)
        content.set_margin_bottom(10)
        content.set_margin_start(14)
        content.set_margin_end(14)

        image = Gtk.Image.new_from_file(

            os.path.join(
                ICON_DIR,
                icon
            )

        )

        image.set_pixel_size(
            20
        )

        label = Gtk.Label(
            label=title
        )

        label.set_xalign(
            0
        )

        label.set_hexpand(
            True
        )

        content.append(
            image
        )

        content.append(
            label
        )

        self.set_child(
            content
        )

    def activate(self):

        self.add_css_class(
            "active"
        )

    def deactivate(self):

        self.remove_css_class(
            "active"
        )

File: rust/preferences/src/widgets/sidebar_item.rs
// ==========================================
// SidebarItem — botón del sidebar con icono + título
// (equivalente a widgets/sidebar_item.py)
// ==========================================

use gtk::prelude::*;

pub struct SidebarItem {
    pub button: gtk::Button,
}

impl SidebarItem {
    pub fn new(icon: &str, title: &str) -> Self {
        let button = gtk::Button::new();
        button.add_css_class("sidebar-item");

        let content = gtk::Box::new(gtk::Orientation::Horizontal, 12);
        content.set_margin_top(10);
        content.set_margin_bottom(10);
        content.set_margin_start(14);
        content.set_margin_end(14);

        let image = gtk::Image::from_file(crate::assets::icon_path(icon));
        image.set_pixel_size(20);
        content.append(&image);

        let label = gtk::Label::builder()
            .label(title)
            .xalign(0.0)
            .build();
        label.set_hexpand(true);
        content.append(&label);

        button.set_child(Some(&content));

        Self { button }
    }

    pub fn widget(&self) -> &gtk::Button {
        &self.button
    }

    pub fn activate(&self) {
        self.button.add_css_class("active");
        self.button.set_state_flags(gtk::StateFlags::CHECKED, false);
    }

    pub fn deactivate(&self) {
        self.button.remove_css_class("active");
        self.button.unset_state_flags(gtk::StateFlags::CHECKED);
    }
}
